services/packagecontrol/packagecontrol.service.js

```diff
@@ -1,148 +1,125 @@
 'use strict'
 
-const LegacyService = require('../legacy-service')
-const { makeBadgeData: getBadgeData } = require('../../lib/badge-data')
+const Joi = require('joi')
+const BaseJsonService = require('../base-json')
 const { metric } = require('../../lib/text-formatters')
-const {
-  downloadCount: downloadCountColor,
-} = require('../../lib/color-formatters')
+const { downloadCount } = require('../../lib/color-formatters')
+const { nonNegativeInteger } = require('../validators')
 
 const keywords = ['sublime', 'sublimetext', 'packagecontrol']
 
-// This legacy service should be rewritten to use e.g. BaseJsonService.
-//
-// Tips for rewriting:
-// https://github.com/badges/shields/blob/master/doc/rewriting-services.md
-//
-// Do not base new services on this code.
-module.exports = class PackageControl extends LegacyService {
-  static get category() {
-    return 'downloads'
-  }
-
-  static get route() {
-    return {
-      base: 'packagecontrol',
-      pattern: ':interval(dm|dw|dd|dt)/:packageName',
-    }
-  }
+const schema = Joi.object({
+  installs: Joi.object({
+    total: nonNegativeInteger,
+    daily: Joi.object({
+      data: Joi.array()
+        .items(
+          Joi.object({
+            totals: Joi.array()
+              .items(nonNegativeInteger)
+              .required(),
+          }).required()
+        )
+        .required(),
+    }).required(),
+  }).required(),
+})
 
-  static get examples() {
-    return [
-      {
-        title: 'Package Control',
-        pattern: 'dm/:packageName',
-        namedParams: { packageName: 'GitGutter' },
-        staticPreview: {
-          label: 'downloads',
-          message: '10k/month',
-          color: 'brightgreen',
-        },
-        keywords,
+function DownloadsForInterval(interval) {
+  const { base, messageSuffix, counter } = {
+    day: {
+      base: 'packagecontrol/dd',
+      messageSuffix: '/day',
+      counter: function(resp) {
```

Comment:
What do you think about calling this `transform`? It could use an arrow function.

Reply:
No technical advantage, just stylistic.

I thought it's what we were using in most places for anonymous functions, though perhaps that is not the case. I'll open a new issue.

Reply:
I tried to find a linter rule to deal with this, but couldn't find one. It seems like a fairly unusual case: a function expression which is a property, and not being used as a callback.

We do have a few function expressions among mostly function declarations. I'll open a PR to tidy up those.